packages/xml-views-validation/test/validators/document/non-unique-id-spec.ts

```diff
@@ -0,0 +1,170 @@
+import { expect } from "chai";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { generateModel } from "@ui5-language-assistant/test-utils";
+import { validateNonUniqueID } from "../../../src/validators/document/non-unique-id";
+import { NonUniqueIDIssue } from "../../../api";
+import {
+  computeExpectedRanges,
+  testValidationsScenario,
+} from "../../test-utils";
+
+describe("the use of non unique id validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  let testNonUniqueIDScenario: (opts: {
+    xmlText: string;
+    assertion: (issues: NonUniqueIDIssue[]) => void;
```

Comment:
Maybe calculate the expected ranges and send it to the assertion? It will save some code which is not test-specific (after we have more than 1 test that checks the ranges)

Reply:
The ranges are used also to assert on `identicalIDsRanges` which is unique to this validation issue. Passing an assertion function and doing calculations is more verbose yet also more versatile/agile for custom flows.

I removed the re-sending of UI5Model / Validation func as those were indeed verbose without cause.

Reply:
I mean to calculate the expected ranges from the xml snippet and send them as a parameter to the `assertion` function (but you will have to call it explicitly instead of passing it to `assertion` of `testValidationsScenario`). Maybe we should even do this generally in `testValidationsScenario`.

Reply:
Ah, yeah could be useful and reduce some LOC